core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/AbstractClusterAction.java

```diff
@@ -29,6 +29,8 @@ protected AbstractClusterAction(Class<P> payloadClass) {
     protected ClusterViewContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> clusterContext, P payload) {
         StackView stack = stackService.getViewByIdWithoutAuth(payload.getResourceId());
         MDCBuilder.buildMdcContext(stack.getClusterView());
+        MDCBuilder.addResourceCrn(stack.getResourceCrn());
+        MDCBuilder.addEnvironmentCrn(stack.getEnvironmentCrn());
```

Comment:
I think it would be better to use `MDCBuilder.buildMdcContext(stack);` first and then the `clusterview` so every field would be populated from the stackview and overwritten by the clusterview if present.

Reply:
I agree, modified it.